packages/toggleWrapper/components/ToggleWrapper.tsx

```diff
@@ -34,80 +34,61 @@ interface LocalToggleWrapperProps extends ToggleWrapperProps {
-  };
-
-  constructor(props: LocalToggleWrapperProps) {
-    super(props);
-
-    this.state = {
-      hasFocus: false
-    };
-
-    this.handleFocus = this.handleFocus.bind(this);
-    this.handleBlur = this.handleBlur.bind(this);
-  }
+const ToggleWrapper = ({
+  id = nextId("toggleWrapper-"),
+  children,
+  "data-cy": dataCy = "toggleWrapper-input",
+  isActive,
+  onFocus,
+  onBlur,
+  ...other
+}: LocalToggleWrapperProps) => {
+  const [hasFocus, setHasFocus] = React.useState<boolean>(false);
 
-  public render() {
-    const {
-      children,
-      id = nextId("toggleWrapper-"),
-      "data-cy": dataCy = "toggleWrapper-input",
-      isActive,
-      ...other
-    } = this.props;
-    const { hasFocus } = this.state;
-    delete other.checked;
-    delete other.className;
-    delete other.onFocus;
-    delete other.onBlur;
+  const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+    setHasFocus(true);
 
-    return (
-      // eslint is giving an error because it can't tell that the value of `type`
-      // will be "radio" or "checkbox".
-      // Passing either string directly makes the error go away
-      /* eslint-disable jsx-a11y/role-supports-aria-props */
-      <label htmlFor={id} className={display("block")}>
-        <input
-          id={id}
-          className={visuallyHidden}
-          checked={isActive}
-          aria-checked={isActive}
-          data-cy={dataCy}
-          onFocus={this.handleFocus}
-          onBlur={this.handleBlur}
-          {...other}
-        />
-        {children({ isActive, hasFocus })}
-      </label>
-      /* eslint-enable jsx-a11y/role-supports-aria-props */
-    );
-  }
+    if (onFocus) {
+      onFocus(e);
+    }
+  };
 
-  private handleFocus(e) {
-    this.setState({ hasFocus: true });
+  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    setHasFocus(false);
 
-    if (this.props.onFocus) {
-      this.props.onFocus(e);
+    if (onBlur) {
+      onBlur(e);
     }
-  }
+  };
 
-  private handleBlur(e) {
-    this.setState({ hasFocus: false });
+  delete other.checked;
+  delete other.className;
 
-    if (this.props.onBlur) {
-      this.props.onBlur(e);
-    }
-  }
-}
+  return (
+    // eslint is giving an error because it can't tell that the value of `type`
+    // will be "radio" or "checkbox".
+    // Passing either string directly makes the error go away
+    /* eslint-disable jsx-a11y/role-supports-aria-props */
+    <label htmlFor={id} className={display("block")}>
+      <input
+        id={id}
+        className={visuallyHidden}
+        checked={isActive}
+        aria-checked={isActive}
+        data-cy={dataCy}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        {...other}
+      />
+      {children({ isActive, hasFocus })}
+    </label>
+    /* eslint-enable jsx-a11y/role-supports-aria-props */
+  );
+};
+
+ToggleWrapper.defaultProps = {
+  type: "checkbox",
+  value: ""
+};
 
 export default ToggleWrapper;
```

Comment:
To retain the functionality of a PureComponent, could you memoize `ToggleWrapper`? We have a similar implementation throughout the codebase.

`export default React.memo(ToggleWrapper);`